edgexfoundry/edgex-go
# fix(all): Update to use user-friendly error message in API responses

fix #3975

This issue is actually located in all services, so fix it in all services.

Signed-off-by: Ginny Guan <ginny@iotechsys.com>







**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I have added unit tests for the new feature or bug fix (if not, why?) only update response error messages
- [ ] I have opened a PR for the related docs change (if not, why?)only update response error messages
  <link to docs PR>

## Testing Instructions

1. build and run docker images for all services from this branch 
2. verify the responses
For example, 
```
For empty profile name,
{"apiVersion":"v2","message":"DeviceProfileBasicInfoRequest.BasicInfo.Name field should not be empty string","statusCode":400}
```
```
For non-existent profile name,
[{"apiVersion":"v2","message":"fail to query device profile by name test","statusCode":404}]
```
## New Dependency Instructions (If applicable)
